### ISSUE TITLE
[Accessibility] Cannot use Shift+Tab as an input key in terminal

### ISSUE THREAD

**Issue by jooyoungseo** [microsoft/vscode#188329]
Type: <b>Bug</b>

## Repro

1. Configure the settings like below:

`settings.json`:

``` json
{
  "terminal.integrated.tabFocusMode": false
}
```

1. Create new terminal via ctrl+`

1. Press `Shift+Tab` from terminal input.

## Current Behavior

Pressing Shift+Tab moves to the terminal buffer.

## Expected Behavior

Shift+Tab should be passed to terminal as an input key. The focus needs to remain in the terminal input field.

We need `Shift+Tab` when cycling back through shell auto-suggestion or ipython completion.

VS Code version: Code - Insiders 1.81.0-insider (c85bf61a82b0c39886b032d2634108782a55c637, 2023-07-19T05:34:51.441Z)
OS version: Windows_NT x64 10.0.22621
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|11th Gen Intel(R) Core(TM) i5-1145G7 @ 2.60GHz (8 x 2611)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.71GB (6.44GB free)|
|Process Argv|--crash-reporter-id b05b88e5-8894-4031-ae34-fa034ebddea9|
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (90)</summary>

Extension|Author (truncated)|Version
---|---|---
android-dev-ext|ade|1.3.2
Bookmarks|ale|13.4.1
openscad|Ant|1.1.1
spellright|ban|3.0.116
zoterolatex|bna|0.4.1
mermaid-markdown-syntax-highlighting|bpr|1.5.2
doxdocgen|csc|1.4.0
dscodegpt|Dan|2.1.14
vscode-markdownlint|Dav|0.51.0
vscode-eslint|dba|2.4.2
vscode-quick-select|dba|0.2.9
vscode-deno|den|3.19.1
gitlens|eam|14.1.1
EditorConfig|Edi|0.16.4
prettier-vscode|esb|9.19.0
vscode-google-translate|fun|1.4.13
codespaces|Git|1.14.12
copilot|Git|1.97.271
copilot-chat|Git|0.5.2023071901
remotehub|Git|0.60.0
vscode-github-actions|git|0.25.8
vscode-pull-request-github|Git|0.68.1
easy-snippet|inu|0.6.3
path-autocomplete|ion|1.24.1
latex-workshop|Jam|9.13.1
lilypond-syntax|jea|0.1.1
scheme|jea|0.2.0
better-cpp-syntax|jef|1.17.2
google-search|kam|0.0.1
vscode-lua-format|Koi|1.3.8
lilypond-formatter|lhl|0.2.3
lilypond-pdf-preview|lhl|0.2.8
lilypond-snippets|lhl|0.1.1
vslilypond|lhl|1.7.3
zotero|mbl|0.1.10
git-graph|mhu|1.30.0
vscode-docker|ms-|1.26.0
black-formatter|ms-|2023.4.1
flake8|ms-|2023.6.0
isort|ms-|2023.11.11921012
python|ms-|2023.12.0
vscode-pylance|ms-|2023.7.30
jupyter|ms-|2023.6.1101941928
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
vscode-jupyter-cell-tags|ms-|0.1.8
vscode-jupyter-slideshow|ms-|0.1.5
remote-containers|ms-|0.301.0
remote-ssh|ms-|0.102.0
remote-ssh-edit|ms-|0.86.0
remote-wsl|ms-|0.80.2
vscode-remote-extensionpack|ms-|0.24.0
azure-repos|ms-|0.36.0
cmake-tools|ms-|1.14.34
cpptools|ms-|1.16.3
cpptools-extension-pack|ms-|1.3.0
js-debug-nightly|ms-|2023.7.1717
remote-repositories|ms-|0.38.1
vscode-github-issue-notebooks|ms-|0.0.129
vscode-selfhost-test-provider|ms-|0.3.15
vscode-serial-monitor|ms-|0.10.0
vsliveshare|ms-|1.0.5873
resourcemonitor|mut|1.0.7
autodocstring|njp|0.6.1
pandocciter|not|0.10.2
shiny-python|Pos|0.1.2
shinyuieditor|pos|0.4.3
quarto|qua|1.90.0
r-debugger|RDe|0.5.4
java|red|1.20.0
vscode-xml|red|0.26.1
r|REd|2.8.1
multi-command|ryu|1.6.0
vscode-deepl|soe|1.0.6
abc-music|sof|0.4.0
lua|sum|3.6.23
latex-utilities|tec|0.4.10
chatgpt|tim|1.1.2
cmake|twx|0.0.17
errorlens|use|3.12.0
intellicode-api-usage-examples|Vis|0.2.7
vscodeintellicode|Vis|1.2.30
vscode-arduino|vsc|0.6.0
vscode-java-debug|vsc|0.52.0
vscode-java-dependency|vsc|0.23.0
vscode-java-pack|vsc|0.25.12
vscode-java-test|vsc|0.39.0
vscode-maven|vsc|0.41.0
markdown-all-in-one|yzh|3.5.1
grammarly|znc|0.22.1

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
vscod805cf:30301675
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30404738
py29gd2263:30784851
vsclangdf:30492506
c4g48928:30535728
dsvsc012:30540252
pynewext54:30618038
pylantcb52:30590116
a9j8j154:30646983
showlangstatbar:30737417
ecj1e332:30687743
pythonfmttext:30716741
pythoncmvfstr:30726892
fixshowwlkth:30771523
hideindicator:30766887
pythongtdpath:30726887
i26e3531:30792625
gsofa:30778558
pythonnosmt12:30779711
pythonidxpt:30768918
pythondjangots:30768917
pythonnoceb:30776497
copilotsettingt:30767686
e537b577:30772214
h0f32768:30792099
synctok:30783813
dsvsc013:30777762
dsvsc014:30777825
diffeditorv2:30786206

```

</details>

**Comment by meganrogge** [comment]
related https://github.com/microsoft/vscode/issues/187875

this also reproduces in stable VS code

**Comment by meganrogge** [comment]
We're going to remove the keybinding altogether as it should enter the buffer when toggle tab moves focus is enabled and not (as designed) without that

**Comment by meganrogge** [comment]
Changing the keybinding could throw users off, though @jooyoungseo is wondering if we can treat this like a widget with results/ input and allow moving between the two with keybindings consistent with #179967

**Comment by meganrogge** [comment]
Shift+tab is still going to the workbench even when I remove that keybinding and have tab moves focus mode off

**Comment by meganrogge** [comment]
We now allow using `alt+f2` or `ctrl/cmd+UpArrow` to go to the terminal accessible buffer instead. 

I think it makes sense to remove `Shift+tab` as a keybinding next iteration if we agree that makes sense @jooyoungseo and @rperez030. The only downside of that would be users might already have `shift+tab` in their muscle memory.

**Comment by rperez030** [comment]
If we cannot resolve the synchronization issue with autocomplete and shift +Tab, I am in favor of removing the key-binding. there are worse things than retraining mussel memory.

**Comment by meganrogge** [comment]
Yes, @jooyoungseo and I just discussed, and we agree it can be removed this iteration.

**Comment by meganrogge** [comment]
We will also remove the tab index

**Comment by alexr00** [comment]
I cannot reproduce the original issue in stable (either 1.81 or 1.78). What are the steps to verify?

**Comment by meganrogge** [comment]
Sorry, to verify, focus a terminal with screen reader mode enabled. Use `shift+tab`. the accessible buffer should not open.

**Comment by meganrogge** [comment]
Guessing we need to do something with tabindex that must've been removed in order to get this to work

**Comment by Tyriar** [comment]
I think it would be either shift+tab needs to be specially handled in xterm.js or we should handle it and cancel the event in the custom key event handler.

**Comment by meganrogge** [comment]
https://github.com/microsoft/vscode/blob/c4a75648e0789ab6c83839165d03403ad24181c6/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L1010

**Comment by meganrogge** [comment]
not sure how this was working before. I dug through commits and tried reverting this one, which didn't help. 

https://github.com/microsoft/vscode/commit/1d0175478873e6ad29033d84925ce5d936032b3c
